### PR TITLE
[RHELC-977] Fix integration tests for pre-check and add assessment-report test case

### PR DESCRIPTION
**Pull request by r0x0d**
This commit tries to fix the rest of the tier0 integration tests that
were failing because of changes made in the pre-check branch.

Signed-off-by: Rodolfo Olivieri <rolivier@redhat.com>

Jira Issues: [RHELC-977](https://issues.redhat.com/browse/RHELC-977)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

File: tests/integration/tier0/assessment-report/test_assessment_report.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        c2r.sendline("y")
+
+        # Assert that we start rollback first
+        assert c2r.expect("Rollback: RHSM-related actions") == 0
+
+        # Then, verify that the analysis report is printed
+        c2r.expect("Pre-conversion analysis report")
+        c2r.expect("No changes needed")
+
+    assert c2r.exitstatus == 0
```
Comment (danmyway) [inline tests/integration/tier0/assessment-report/test_assessment_report.py:68]: I am sort of confused by the docstring, but I believe, we can also verify the unwanted headers are not present.
I would also assert for the conversion report being printed.
```suggestion
@pytest.mark.test_successfull_report
def test_successfull_report(convert2rhel):
    """
    Test if the assessment report contains the following header: Success header.

    And does not contain: Error header, skip header, success header.
    """
    with convert2rhel(
        "--no-rpm-va --serverurl {} --username {} --password {} --pool {} --debug".format(
            env.str("RHSM_SERVER_URL"),
            env.str("RHSM_USERNAME"),
            env.str("RHSM_PASSWORD"),
            env.str("RHSM_POOL"),
        )
    ) as c2r:
        # We need to get past the data collection acknowledgement.
        c2r.expect("Continue with the system conversion?")
        c2r.sendline("y")

        # Assert that we start rollback first
        assert c2r.expect("Rollback: RHSM-related actions") == 0

        # Then, verify that the analysis report is printed
        assert c2r.expect("Pre-conversion analysis report", timeout=600) == 0
        
        # Verify that only success header printed out, assert if error or skip header appears
        c2r_report_header_index = c2r.expect(["No changes needed", "Must fix before conversion", "Could not be checked due to other failures"], timeout=300)
        if c2r_report_header_index == 0:
            pass
        elif c2r_report_header_index == 1:
            assert AssertionError("Error header in the analysis report.")
        elif c2r_report_header_index == 2:
            assert AssertionError("Skip header in the analysis report.")

    assert c2r.exitstatus == 0
```

File: tests/integration/tier0/assessment-report/test_assessment_report.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        # Error header first
+        c2r.expect("Must fix before conversion")
+        c2r.expect("SUBSCRIBE_SYSTEM.UNKNOWN_ERROR: Unable to register the system")
+
+        # Skip header
+        c2r.expect("Could not be checked due to other failures")
+        c2r.expect("ENSURE_KERNEL_MODULES_COMPATIBILITY.SKIP: Skipped")
+
+        # Success header
+        c2r.expect("No changes needed")
```
Comment (danmyway) [inline tests/integration/tier0/assessment-report/test_assessment_report.py:37]: I would assert for the pre-conversion analysis report and the headers as those are pretty much the focus points of the test.
```suggestion
        # Then, verify that the analysis report is printed
        assert c2r.expect("Pre-conversion analysis report", timeout=600) == 0

        # Error header first
        assert c2r.expect("Must fix before conversion", timeout=600) == 0
        c2r.expect("SUBSCRIBE_SYSTEM.UNKNOWN_ERROR: Unable to register the system")

        # Skip header
        assert c2r.expect("Could not be checked due to other failures", timeout=600) == 0
        c2r.expect("ENSURE_KERNEL_MODULES_COMPATIBILITY.SKIP: Skipped")

        # Success header
        assert c2r.expect("No changes needed", timeout=600) == 0
```

File: tests/integration/tier1/remove-all-submgr-pkgs/test_no_submgr_pkg_installed.py [hunk 3]
```diff
@@ -24,12 +20,10 @@ def test_no_sub_manager_installed(shell, convert2rhel):
             env.str("RHSM_POOL"),
         )
     ) as c2r:
-        while prompt_amount > 0:
-            c2r.expect("Continue with the system conversion?")
-            c2r.sendline("y")
-            prompt_amount -= 1
+        c2r.expect("Continue with the system conversion?")
+        c2r.sendline("y")
```
Comment (danmyway) [inline tests/integration/tier1/remove-all-submgr-pkgs/test_no_submgr_pkg_installed.py:24]:
```suggestion
        # We need to get past the data collection acknowledgement.
        c2r.expect("Continue with the system conversion?")
        c2r.sendline("y")
```

File: tests/integration/tier0/assessment-report/main.fmf [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+tag+:
+    - pre-assessment-report
+
+/failures_and_skips_in_report:
+    summary+: |
+        Verify that some actions failures and skips appeared in the report.
+    description+: |
+        Verify that the conversion will proceed until the point of no return
+        even with failures and skips, and then, output the report to the user.
```
Comment (danmyway) [inline tests/integration/tier0/assessment-report/main.fmf:15]: Small addition to the description and trying to keep the summary concise.
```suggestion
    summary+: |
        Failed registration skips checks
    description+: |
        Verify that the conversion will proceed until the point of no return
        even with failures and skips, and the report is printed to the user
        with expected report headers.
        Intentionally pass wrong registration credentials to fail. 
```

File: tests/integration/tier0/assessment-report/main.fmf [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        - failed-report
+    test: |
+        pytest -svv -m test_failures_and_skips_in_report
+
+/successfull_report:
+    summary+: |
+        Verify that the report was successfull.
+    description+: |
+        Verify that the conversion will proceed until the point of no return
+        and then, output the report to the user.
```
Comment (danmyway) [inline tests/integration/tier0/assessment-report/main.fmf:28]:
```suggestion
    summary+: |
        All reported checks successful
    description+: |
        Simulate the ideal scenario with all checks passing.
        Verify that the conversion will proceed until the point of no return
        and then, output the report to the user.
        No skip or error header is present in the report.
```

File: tests/integration/tier0/assessment-report/main.fmf [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    environment+:
+        CONVERT2RHEL_EXPERIMENTAL_ANALYSIS: 1
+    tag+:
+        - success-report
+    test: |
+        pytest -svv -m test_successfull_report
```
Comment (danmyway) [inline tests/integration/tier0/assessment-report/main.fmf:34]: Small note, I would just append the markers `test_successfull_report` and `test_failures_and_skips_in_report` to `pytest.ini`

**Review by danmyway — CHANGES_REQUESTED** [review]
Looks great @r0x0d !
Thanks for all the hard work on this, I really appreciate it!

File: tests/integration/tier0/assessment-report/main.fmf (hunk 6 above)
Comment (r0x0d) [reply tests/integration/tier0/assessment-report/main.fmf]: done.

[the remark this answers is not in the document]
Comment (r0x0d) [reply tests/integration/tier0/assessment-report/main.fmf]: updated that manually.

**Review by abadger — APPROVED** [review]
To the extent that I am competent at evaluating integration tests, this looks good to me.

**Review by danmyway — APPROVED** [review]
Looks great! Thanks for working on this!

**Pull request merged**